### PR TITLE
Docs: Use `instanceof` in `::Range` pattern description

### PR DESCRIPTION
**Pull request by RasmusWL**
Had to change the example away from Go, since their libs have not been
updated to use `instanceof` yet (but I'll make a PR from them shortly)

### CODE REVIEW

File: docs/ql-design-patterns.md [hunk 1]
```diff
@@ -56,22 +52,18 @@ module MySpecialExpr {
...
+Let's use an example from the Python libraries: https://github.com/github/codeql/blob/46751e515c40c6b4c9b61758cc840eec1894a624/python/ql/lib/semmle/python/Concepts.qll#L601-L683
 
-`EscapeFunction`, as the name suggests, models various APIs that escape meta-characters. It has a member-predicate `kind()` that tells you what sort of escaping the modelled function does. For example, if the result of that predicate is `"js"`, then this means that the escaping function is meant to make things safe to embed inside JavaScript.
-`EscapeFunction::Range` is subclassed to model various APIs, and `kind()` is implemented accordingly.
-But we can also subclass `EscapeFunction` to, as in the above example, talk about all JS-escaping functions.
+`Escaping`, as the name suggests, models various APIs that escape meta-characters. It has a member-predicate `getKind()` that tells you what sort of escaping the modelled function does. For example, if the result of that predicate is `"html"`, then this means that the escaping function is meant to make things safe to embed inside HTML.
```
[the remark this answers is not in the document]
Comment (RasmusWL) [reply docs/ql-design-patterns.md:61]: true, thanks :+1:

File: docs/ql-design-patterns.md [hunk 2]
```diff
@@ -56,22 +52,18 @@ module MySpecialExpr {
...
+Let's use an example from the Python libraries: https://github.com/github/codeql/blob/46751e515c40c6b4c9b61758cc840eec1894a624/python/ql/lib/semmle/python/Concepts.qll#L601-L683
 
-`EscapeFunction`, as the name suggests, models various APIs that escape meta-characters. It has a member-predicate `kind()` that tells you what sort of escaping the modelled function does. For example, if the result of that predicate is `"js"`, then this means that the escaping function is meant to make things safe to embed inside JavaScript.
-`EscapeFunction::Range` is subclassed to model various APIs, and `kind()` is implemented accordingly.
-But we can also subclass `EscapeFunction` to, as in the above example, talk about all JS-escaping functions.
+`Escaping`, as the name suggests, models various APIs that escape meta-characters. It has a member-predicate `getKind()` that tells you what sort of escaping the modeled function does. For example, if the result of that predicate is `"html"`, then this means that the escaping function is meant to make things safe to embed inside HTML.
+`Escaping::Range` is subclassed to model various APIs, and `kind()` is implemented accordingly.
```
Comment (yoff) [inline docs/ql-design-patterns.md:62]:
```suggestion
`Escaping::Range` is subclassed to model various APIs, and `kind()` is implemented accordingly. This typically happens in library models.
```

File: docs/ql-design-patterns.md [hunk 3]
```diff
@@ -56,22 +52,18 @@ module MySpecialExpr {
...
+But we can also subclass `Escaping` to, as in the above example, talk about all HTML-escaping functions.
```
Comment (yoff) [inline docs/ql-design-patterns.md:63]:
```suggestion
But we can also subclass `Escaping`, as in the above example, where `HtmlEscaping` represents all HTML-escaping functions.
```

**Review by yoff — CHANGES_REQUESTED** [review]
I know it is beyond the current edit, but I feel there is an opportunity to clarify a bit, so I made suggestions to that effect.

File: docs/ql-design-patterns.md (hunk 2 above)
Comment (RasmusWL) [reply docs/ql-design-patterns.md:62]:
```suggestion
`Escaping::Range` is subclassed to model various APIs, and `kind()` is implemented accordingly (this typically happens in library models).
```

**Pull request merged**